p2p/src/peers.rs

```diff
@@ -605,7 +598,12 @@ impl ChainAdapter for Peers {
 				"Received a bad block {} from  {}, the peer will be banned",
 				hash, peer_info.addr,
 			);
-			self.ban_peer(peer_info.addr, ReasonForBan::BadBlock);
+			self.ban_peer(peer_info.addr, ReasonForBan::BadBlock)
+				.map_err(|e| {
+					let err: chain::Error =
```

Comment:
Here and below - perhaps just return error `chain::ErrorKind::Other(format!("ban peer error :{:?}", e)).into()` instaead of defining a temp var and returnning it?

Reply:
I have no chance otherwise Rust complains (cannot infer type thing).

Reply:
Unfortunately cannot be done. Rust need type annotations for that one.